### PR TITLE
[EOC-435] Adding, archiving, restoring an item don't change list metadata

### PR DESCRIPTION
**Pull request by alemikolo**
Because of refactoring event emitting for items, I had to merge those changes to EOC-435 and  I had to make some changes in the previous implementation.

### CODE REVIEW

File: server/common/utils/helpers.js [hunk 1]
```diff
@@ -137,17 +137,31 @@ const responseWithItems = (userId, items) =>
...
     ...rest,
     authorId,
     authorName,
     isArchived,
     isVoted: checkIfArrayContainsUserId(voterIds, userId),
     votesCount: voterIds.length
   };
+
+  if (editor) {
+    const { displayName } = editor;
```
Comment (adamklepacz) [inline server/common/utils/helpers.js]: line of a break

File: server/sockets/list.js [hunk 2]
```diff
@@ -143,63 +146,48 @@ const updateItemState = (socket, itemClientLocks) => {
...
-                const { socketId } = listClients.get(viewerIdAsString);
+  return List.findById(listId)
+    .lean()
+    .populate('items.authorId', 'displayName')
+    .populate('items.editedBy', 'displayName')
+    .exec()
+    .then(doc => {
+      if (doc) {
+        const { items, viewersIds } = doc;
+        const indexOfItem = items.findIndex(
```
Comment (adamklepacz) [inline server/sockets/list.js]: `itemIndex`

File: server/sockets/list.js [hunk 3]
```diff
@@ -143,63 +146,48 @@ const updateItemState = (socket, itemClientLocks) => {
...
-  listClients,
-  viewersIds
+  cohortClients,
+  dashboardClients,
+  listClients
 ) => data => {
-  const { listId, userId, itemId } = data;
-  const sanitizedUserId = sanitize(userId);
-
-  return User.findById(sanitizedUserId).then(user => {
```
Comment (adamklepacz) [inline server/sockets/list.js]: does it update edited by property on front end now??

Comment (alemikolo) [reply server/sockets/list.js]: Yes it updates this prop

File: server/controllers/list.js [hunk 4]
```diff
@@ -1133,15 +1137,13 @@ const updateListItem = (req, res) => {
         throw new NotFoundException();
       }
 
-      const { items, viewersIds } = doc;
+      const { items } = doc;
       const itemToUpdate = items.id(sanitizedItemId);
-      let data;
       itemToUpdate.editedBy = userId;
 
       if (description !== undefined) {
```
Comment (adamklepacz) [inline server/controllers/list.js]: plesae replace all !== undefined with isDefined helper on this branch.

File: server/sockets/list.js [hunk 5]
```diff
@@ -210,9 +198,15 @@ const addComment = io => data => {
...
+const cloneItem = (
+  io,
+  cohortClients,
+  dashboardClients,
+  listClients,
+  viewersIds
+) => data => {
   const { userId, item, listId } = data;
-  const dataToSend = { ...data, ...item };
+  const dataToSend = { listId, item };
```
Comment (Ciunkos) [inline server/sockets/list.js]: Consider sorting this.

**Pull request merged**